packages/flutter_tools/lib/src/build_system/targets/native_assets.dart

```diff
@@ -56,6 +56,14 @@ class NativeAssets extends Target {
 
     final Uri projectUri = environment.projectDir.uri;
     final FileSystem fileSystem = environment.fileSystem;
+
+    final List<Uri> dependencies;
+
+    if (targetPlatform == TargetPlatform.darwin && environment.defines[kDarwinNativeAssets] == 'false') {
```

Comment:
This will affect users who run `flutter run -d macos` or `flutter build macos`, right? We have far more developers running these two commands than `flutter build bundle`.

I'm sorry, I think when I said "this change will affect other targets", it was poorly worded. What I meant was, it will affect not just developers trying to build a bundle, but all apps or modules or plugins.

Reply:
does this bug only affect mac users?

Reply:
The execution of `flutter run -d macos` and `flutter build macos` goes through this function. However, the new flag `environment.defines[kDarwinNativeAssets]` is only set to false from the bundle_builder below. So it would only affect users who run `flutter build bundle --target-platform darwin`.

And, yes, this bug only affects mac users.